### PR TITLE
llvm: address v11-dev compatibility issues

### PR DESCRIPTION
**Pull request by vtjnash**
LLVM is unabashedly creating more nuisance renaming changes than covered here (https://groups.google.com/forum/#!topic/llvm-dev/DZRKP9qriZU), so the last known good commit is e252293d0629066225228596a9ef9d397513eb99. But this is a start towards staying functional.

### CODE REVIEW

File: src/llvm-version.h [hunk 1]
```diff
@@ -15,3 +15,5 @@
 #if JL_LLVM_VERSION < 100000
 #define Align(a) (a)
 #endif
+
+#define LLVM_DISABLE_ABI_BREAKING_CHECKS_ENFORCING 0
```
Comment (Keno) [inline src/llvm-version.h]: Why is this necessary now?

Comment (vtjnash) [reply src/llvm-version.h]: I'm tired of carrying the patch to upstream

Comment (Keno) [reply src/llvm-version.h]: Which LLVM patch is that? I thought these checks were useful to avoid accidentally linking the wrong LLVM ABI.

Comment (vtjnash) [reply src/llvm-version.h]: careful--I'm disabling not disabling the checks here. https://github.com/JuliaLang/julia/blob/6aa0d924ff5431318a6c79c7c9ce3a87297170d0/deps/patches/llvm-6.0-DISABLE_ABI_CHECKS.patch

Comment (Keno) [reply src/llvm-version.h]: > careful--I'm disabling not disabling the checks here.

You are if the patch is applied though, right?

Comment (vchuravy) [reply src/llvm-version.h]: That patch is morally wrong and apparently I used to not understand C pre processor flags. I am happy with dropping that patch.
